src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  let currentState = { ...state };
```

Comment:
[NAMING] - The name 'currentState' is a bit misleading as it suggests it is a single state, whereas it actually represents the state after each action. Consider renaming it to something like 'stateCopy' to better reflect its purpose.